tests/unit/casmutils/coordinate.cpp

```diff
@@ -1,25 +1,66 @@
 #include <casmutils/xtal/coordinate.hpp>
+#include <casmutils/xtal/lattice.hpp>
 #include <gtest/gtest.h>
 #include <memory>
 #include <utility>
 
-class CoordinateTest : public testing::Test
-{
+class CoordinateTest : public testing::Test{
 protected:
-    void SetUp() override
-    {
+    void SetUp() override{
         Eigen::Vector3d raw_coord(0.1, 0.2, 0.3);
+        Eigen::Matrix3d lattice;
+        lattice << 0,0.5,0.5,
+                0.5,0,0.5,
+                0.5,0.5,0;
+        
         coord0_ptr.reset(new rewrap::Coordinate(raw_coord));
         coord1_ptr.reset(new rewrap::Coordinate(raw_coord(0), raw_coord(1), raw_coord(2)));
+        lattice_ptr.reset(new rewrap::Lattice(lattice));
     }
-
     // Use unique pointers because Coordinate has no default constructor
     std::unique_ptr<rewrap::Coordinate> coord0_ptr;
     std::unique_ptr<rewrap::Coordinate> coord1_ptr;
+    std::unique_ptr<rewrap::Lattice> lattice_ptr;
 };
 
 TEST_F(CoordinateTest, Construct) { EXPECT_EQ(coord0_ptr->cart(), coord1_ptr->cart()); }
 
+TEST_F(CoordinateTest, FracRetrieve){
+    rewrap::Lattice &lattice_rewrap = *lattice_ptr;
+    Eigen::Matrix3d lattice;
```

Comment:
The use of ```lattice_ptr``` here allows access to the ```casmutils::xtal::Lattice``` constructed in the test frame. You seem to be recreating a matrix that you made in the test frame which is code duplication. Consider putting the raw ```Eigen::Matrix3d``` in the test frame. Also please use variable names that help discern type. i.e. lattice_matrix for ```Eigen::Matrix3d``` and lattice for ```casmutils::xtal::Lattice```

Reply:
Done